tests/unit/linking/test_link.py

```diff
@@ -529,6 +529,115 @@ def test_missing_field_points_exceed_max_missingness_fraction(
         # low-value missing field, but fails user's overall completeness constraint
         assert matches == [False, False]
 
+    def test_no_match_one_suffix_one_not(
+        self,
+        session,
+        default_algorithm,
+        patients: list[schemas.PIIRecord]
+    ):
+        # Make a deep copy of the first patient, give it a suffix
+        patients = [patients[0]]
+        duplicate = copy.deepcopy(patients[0])
+        duplicate.external_id = str(uuid.uuid4())
+        duplicate.name[0].suffix = ["Sr"]
+        patients.append(duplicate)
+
+        matches: list[bool] = []
+        mapped_patients: dict[str, int] = collections.defaultdict(int)
+        for data in patients[:2]:
+            (_, person, results, _) = link.link_record_against_mpi(data, session, default_algorithm)
+            matches.append(bool(person and results))
+            mapped_patients[person.reference_id] += 1
+
+        # First patient inserted into empty MPI, no match
+        # Second patient fails eval in first pass and blocking in second
+        # due to suffix
+        assert matches == [False, False]
+
+    def test_no_match_same_name_diff_suffixes(
+        self,
+        session,
+        default_algorithm,
+        patients: list[schemas.PIIRecord]
+    ):
+        '''
+        NOTE: This catches the Jr/Sr edge case of a parent and child
+        living at the same address with the same first name. It even
+        goes a step farther and gives them the same birthday.
```

Comment:
dang, incredible!

Reply:
Prepending suffix was so helpful. The difference in J vs S makes it auto-fail blocking in pass 2, so address and birthdate never have a chance to come up, and in pass 1, the letter difference at the first character really tanks JaroWinkler from the start because earlier typos are more heavily weighted. It was a great idea on your part!